src/Roave/SecurityAdvisories/Version.php

```diff
@@ -4,24 +4,36 @@
 
 namespace Roave\SecurityAdvisories;
 
-/**
- * A simple version, such as 1.0 or 1.0.0.0 or 2.0.1.3.2
- */
+use Composer\Semver\VersionParser;
+
 final class Version
 {
-    const VALIDITY_MATCHER = '/^(?:\d+\.)*\d+$/';
+    private const STABILITY_TAIL =  '[._-]?'.
+                                    '(?:(stable|beta|rc|alpha|patch)((?:[.-]?\d+)+)?)?'.
+                                    '([.-]?dev)?';
+
+    private const VALIDITY_MATCHER = '/^(?:\d+\.)*\d+'.self::STABILITY_TAIL.'$/';
+
+    /**
+     * @var string
+     */
+    private $version;
 
     /**
-     * @var string[]
+     * @var VersionParser
      */
-    private $versionNumbers;
+    private static $versionParser;
 
     /**
-     * @param int[] $versionNumbers
+     * @param string $version
      */
-    private function __construct(array $versionNumbers)
+    private function __construct(string $version)
     {
-        $this->versionNumbers = $versionNumbers;
+        if (self::$versionParser == null) {
+            self::$versionParser = new VersionParser();
```

Comment:
As mentioned above, the `VersionParser` cannot cope with `'0.1.2.3.4'`, which breaks the invariants defined by tests below